storage/redis_cluster.go

```diff
@@ -666,25 +661,16 @@ func (r *RedisCluster) StartPubSubHandler(channel string, callback func(interfac
 		return errors.New("Redis connection failed")
 	}
 
-	pubsub := client.Subscribe(r.RedisController.ctx, channel)
+	pubsub := client.Subscribe(ctx, channel)
 	defer pubsub.Close()
 
 	for {
-		msg, err := pubsub.Receive(r.RedisController.ctx)
-		if err != nil {
-			log.Error("Error while receiving pubsub message:", err)
-			return err
-		}
-		switch v := msg.(type) {
-		case *redis.Message:
-			callback(v)
-
-		case *redis.Subscription:
-			callback(v)
```

Comment:
Was only used in tests, removed now

Reply:
We should explicitly add this section to QA testing. Multi note setup with distributed rate limiting, where instances gets dynamically added/removed from the cluster.